### DISCUSSION TITLE
GRPC query support

### DISCUSSION THREAD

**Discussion by pdeaudney** [stepci/stepci#19] [Ideas]
Would be great to be able to also query GRPC api endpoints.

**Comment by mishushakov** [comment]
Hello, @pdeaudney

Thanks for your suggestion! This is something we hear a lot and have already added to our roadmap!
You can subscribe to this issue: https://github.com/stepci/runner/issues/52 to get updates

Also, can I ask you whether you have any issues or anything you wish was improved when you're developing gRPC services?

**Comment by mishushakov** [comment]
For all our gRPC fans out there - we have just published a type-safe gRPC client for Node, which is fun to work with. It has a cool name also!

https://github.com/stepci/cool-grpc

**Comment by mishushakov** [comment]
Hey there,
We just added gRPC support to Step CI 😁

Demo: https://stepci.com
Example: https://github.com/stepci/stepci/blob/main/examples/grpc.yml
Reference: https://github.com/stepci/stepci/blob/main/docs/workflow-syntax.md#teststeststepsstepgrpc

Let us know what you think! 😉

**Comment by pdeaudney** [comment]
Where I work has chosen to structure their proto definitions for company proto files under a path such as 
`./protos/s12/accounts/user/v1/user.proto`

In this `user.proto` we have imports to vendored 3rd party protos that are stored in a directory prefixed with `third_party`. 
```
import "google/api/annotations.proto";
import "google/api/visibility.proto";
```

The pb building toolchain in use is [buf](https://buf.build/) and that supports us having our own proto files separated out from third_party/vendored protos in the same folder structure. For example our `buf.work.yaml` is setup as 
```
version: v1
directories:
  - protos
  - third_party
```
I am not sure if this is a widely used convention.

Currently when I try to test a simple RPC from `user.proto` that imports the third_party google protos the path loading stepci wants is 
```
ENOENT: no such file or directory, open 'protos/s12/accounts/user/v1/google/api/annotations.proto'
```

It would be good if I could pass in some directories to the proto file section and have stepci parse all of them. Although I am not sure how feasible that is (i'm not a expert in protobuf or its usage).

**Reply by mishushakov** [reply]
Hey 😄
This is indeed a valid point. We will be using ezy's universal gRPC client, which should solve it.

Please see the discussion below:
https://github.com/stepci/stepci/discussions/80

Check out [ezy](https://github.com/getezy/ezy) gRPC Client, it's awesome 😉

**Comment by mishushakov** [comment]
Hey @pdeaudney, we just improved our gRPC integration to work with proto-loader in runner version 1.11.0
